From a8ce803ac7a535238464a54d90a985a246ba7a25 Mon Sep 17 00:00:00 2001
From: Rong Bao <webmaster@csmantle.top>
Date: Thu, 27 Aug 2026 02:16:25 +0000
Subject: [PATCH 07/19] Bug 2066959 - Enable -fstack-clash-protection on
 riscv64 Clang>=22.1.8. r=glandium!

This feature was merged into Clang's release/22.x on Jun 12, 2026 and
was released with Clang 22.1.8. It was also shipped with the main
branch.

Link: https://github.com/llvm/llvm-project/pull/195456
 ("[CodeGen][RISCV] Inline stack probes immediately after allocateStack
 in eliminateCallFramePseudoInstr")
Link: https://github.com/llvm/llvm-project/pull/202882 ("release/22.x
 v2: [CodeGen][RISCV] Inline stack probes immediately after allocateStack
 in eliminateCallFramePseudoInstr (#195456)")

Differential Revision: https://phabricator.services.mozilla.com/D321753
---
 build/moz.configure/toolchain.configure | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index da055a84271f4cb338638cd4bd9972b13bf7a158..4a416b5476f802c6322271a5229e0c97867055c4 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -2724,7 +2724,9 @@ def security_hardening_cflags(
                 and target.os not in ("WINNT", "OSX", "OpenBSD", "iOS")
                 and ((c_compiler.version >= "11.0.1" and target.cpu in ("x86", "x86_64", "ppc64", "s390x"))
                     or
-                     (c_compiler.version >= "18.1.0" and target.cpu == "aarch64"))
+                     (c_compiler.version >= "18.1.0" and target.cpu == "aarch64")
+                    or
+                     (c_compiler.version >= "22.1.8" and target.cpu == "riscv64"))
             ):
                 flags.append("-fstack-clash-protection")
                 ldflags.append("-fstack-clash-protection")
-- 
2.55.0

